### PR TITLE
[IMP] runbot: detach odoo command from docker_run

### PR DESCRIPTION
**Pull request by d-fence**
When the docker_run function is called, the odoo command is decorated
with a pip command to install required packages.
This pollute the docker_run function if a runbot job_ method wants to
use docker for something else that starting an odoo instance (like
pg_dump) for example.

With this commit, command modification is made in an optional helper
function named build_odoo_cmd.

the docker_run function now needs the command to run as a string instead
of a list of odoo cmd and its parameters.

### CODE REVIEW

**Comment by d-fence** [comment]
As a side note:
I encountered a problem while testing this PR. When the I use the "psql" command instead of odoo-bin like [here](https://github.com/odoo/runbot/pull/112/files#diff-a781692ae7da61310cb61986595e67a2R172), if the postgres version on the host is different from container, the command fails. But it's not directly linked to the PR code.

File: runbot/container.py [hunk 1]
```diff
@@ -153,13 +161,24 @@ def tests(args):
...
+    if args.dump:
+        _logger.info("Testing pg_dump")
+        logfile = os.path.join(args.build_dir, 'logs', 'logs-pg_dump.txt')
+        container_name = 'odoo-container-test-pg_dump-%s' % datetime.datetime.now().microsecond
+        docker_pg_dump_cmd = 'cd /data/build/datadir && pg_dump -U %s -f db_export.sql %s' % (os.getlogin(), args.db_name)
+        docker_run(docker_pg_dump_cmd, logfile, args.build_dir, container_name)
+        time.sleep(1)
+        while docker_is_running(container_name):
+            time.sleep(10)
+            _logger.info("Waiting for %s to stop", container_name)
```
Comment (xmo-odoo) [inline runbot/container.py]: Maybe these wait loops could just use [docker wait](https://docs.docker.com/engine/reference/commandline/wait/) instead?

Comment (d-fence) [reply runbot/container.py]: Great, I was not aware of this docker command. Thanks

Comment (d-fence) [reply runbot/container.py]: On the other hand, it tests the "docker_is_running" function

**Pull request merged**